contrib/hack/migration-check.sh

```diff
@@ -0,0 +1,168 @@
+        if [[ "${class}" = "${externalName}" ]]; then
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+function checkIfClassExistForInstance(){
+    serviceClassesNames=$(kubectl get serviceclasses --all-namespaces -ojsonpath="{.items[*].spec.externalName}")
+    clusterServiceClassesNames=$(kubectl get clusterserviceclasses -ojsonpath="{.items[*].spec.externalName}")
+
+    for className in $(kubectl get serviceinstances --all-namespaces -ojsonpath="{.items[*].status.userSpecifiedClassName}")
+    do
+        type=$(echo "${className}" | cut -d'/' -f1)
+        externalName=$(echo "${className}" | cut -d'/' -f2)
+
+        set +o errexit
+        if [[ "${type}" = "ClusterServiceClass" ]]; then
+            checkIfClassExist "${externalName}" "${clusterServiceClassesNames}"
+            if [[ $? -eq 1 ]]; then
+                echo "${className} not exist in the cluster for the ServiceInstance anymore"
+                EXIT_CODE=1
+            fi
+        fi
+        if [[ "${type}" = "ServiceClass" ]]; then
+            checkIfClassExist "${externalName}" "${serviceClassesNames}"
+            if [[ $? -eq 1 ]]; then
+                echo "${className} not exist in the cluster for the ServiceInstance anymore"
+                EXIT_CODE=1
+            fi
+        fi
+    done
+    set -o errexit
+}
+
+#
+# Check if any class/plans were removed from broker's catalog
+#
+CSC=$(kubectl get clusterserviceclasses -ojsonpath="{.items[*].metadata.deletionTimestamp}")
+if [[ -n ${CSC} ]]; then
+    echo "There are ClusterServiceClasses with deletionTimestamp set"
+    EXIT_CODE=1
+fi
+for status in $(kubectl get clusterserviceclasses -ojsonpath="{.items[*].status.removedFromBrokerCatalog}")
+do
+if [[ -n "${status}" ]] && ${status}; then
+    echo "There are removedFromBrokerCatalog ClusterServiceClasses"
+    EXIT_CODE=1
+fi
+done
+
+SC=$(kubectl get serviceclasses --all-namespaces -ojsonpath="{.items[*].metadata.deletionTimestamp}")
+if [[ -n ${SC} ]]; then
+    echo "There are ServiceClasses with deletionTimestamp set"
+    EXIT_CODE=1
+fi
+for status in $(kubectl get serviceclasses --all-namespaces -ojsonpath="{.items[*].status.removedFromBrokerCatalog}")
+do
+if [[ -n "${status}" ]] && ${status}; then
+    echo "There are removedFromBrokerCatalog ServiceClasses"
+    EXIT_CODE=1
+fi
+done
+
+CSP=$(kubectl get clusterserviceplans -ojsonpath="{.items[*].metadata.deletionTimestamp}")
+if [[ -n ${CSP} ]]; then
+    echo "There are ClusterServicePlans with deletionTimestamp set"
+    EXIT_CODE=1
+fi
+for status in $(kubectl get clusterserviceplans -ojsonpath="{.items[*].status.removedFromBrokerCatalog}")
+do
+if [[ -n "${status}" ]] && ${status}; then
+    echo "There are removedFromBrokerCatalog ClusterServicePlans"
+    EXIT_CODE=1
+fi
+done
+
+SP=$(kubectl get serviceplans --all-namespaces -ojsonpath="{.items[*].metadata.deletionTimestamp}")
+if [[ -n ${SP} ]]; then
+    echo "There are ServicePlans with deletionTimestamp set"
```

Comment:
please make it consistent 
in the case of service brokers you are saying:
`"There are being deleted ServiceBrokers"` 

so please have always the same message, and better will be described what is going on instead of saying that you have `deletionTimestamp` set becasue it is an implementation details and service catalog users may not know that.

Reply:
done